packages/cli/src/utils/scaffold.js

```diff
@@ -79,42 +126,248 @@ const writeTemplateFile = async (
-  // ensure a clean access
-  delete require.cache[require.resolve(entryFilePath)];
+/**
+ *
+ * @param {Object} options
+ * @param {string} options.indexFileResolved - The App's entry point (index.js/ts)
+ * @param {string} options.actionRelativeImportPath - The path to import the new action with (relative to the index)
+ * @param {string} options.actionImportName - The name of the import, i.e the action key converted to camel_case
+ * @param {ActionType} options.actionType - The type of action, e.g. 'trigger'
+ */
+const updateEntryFileTs = async ({
+  indexFileResolved,
+  actionRelativeImportPath,
+  actionImportName,
+  actionType,
+}) => {
+  let codeStr = (await readFile(indexFileResolved)).toString();
+  const originalCodeStr = codeStr; // untouched copy in case we need to bail
 
-  // this line fails if `npm install` hasn't been run, since core isn't present yet.
-  const rewrittenIndex = require(entryFilePath);
-  return Boolean(_.get(rewrittenIndex, [plural(actionType), newActionKey]));
+  codeStr = importActionInTsApp(
+    codeStr,
+    actionImportName,
+    actionRelativeImportPath
+  );
+  codeStr = registerActionInTsApp(
+    codeStr,
+    plural(actionType),
+    actionImportName
+  );
+  await writeFile(indexFileResolved, codeStr);
+  return originalCodeStr;
+};
+
+/**
+ *
+ * Prepare everything needed to define what's happening in a scaffolding
+ * operation.
+ *
+ * @param {Object} options
+ * @param {ActionType} options.actionType - the action type
+ * @param {string} options.noun - the noun for the action
+ * @param {'js' | 'ts'} options.language - the language of the project
+ * @param {string} options.indexFileLocal - the App's entry point (index.js/ts)
+ * @param {string} options.actionDirLocal - where to put the new action
+ * @param {string} options.testDirLocal - where to put the new action's test
+ * @param {boolean} options.includeIntroComments - whether to include comments in the template
+ * @param {boolean} options.preventOverwrite - whether to force overwrite
+ *
+ * @returns {ScaffoldContext}
+ */
+const createScaffoldingContext = ({
+  actionType,
+  noun,
+  language,
+  indexFileLocal,
+  actionDirLocal,
+  testDirLocal,
+  includeIntroComments,
+  preventOverwrite,
+}) => {
+  const key = nounToKey(noun);
+
+  const indexFileResolved = path.join(process.cwd(), indexFileLocal);
+  const actionFileResolved = `${path.join(
+    process.cwd(),
+    actionDirLocal,
+    key
+  )}.${language}`;
+  const actionFileResolvedStem = path.join(process.cwd(), actionDirLocal, key);
+  const actionFileLocal = `${path.join(actionDirLocal, key)}.${language}`;
+  const actionFileLocalStem = path.join(actionDirLocal, key);
+  const testFileResolved = `${path.join(
+    process.cwd(),
+    testDirLocal,
+    key
+  )}.test.${language}`;
+  const testFileLocal = `${path.join(testDirLocal, key)}.${language}`;
+  const testFileLocalStem = path.join(testDirLocal, key);
```

Comment:
```suggestion
  const testFileResolved = `${path.join(
    cwd,
    testDirLocal,
    key
  )}.test.${language}`;
  const testFileLocalStem = path.join(testDirLocal, key);
  const testFileLocal = `${testFileLocalStem}.${language}`;
```

Reply:
I'm preferring to keep them flat, not depending on each other, and in the definition order.